Panic in markets code

User pasted this stack trace in slack:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1195cc8]
goroutine 1 [running]:
sync.(*RWMutex).RLock(...)
        /usr/lib/golang/src/sync/rwmutex.go:48
github.com/filecoin-project/go-fil-markets/storagemarket/impl/storedask.(*StoredAsk).GetAsk(0x0, 0xc00052245b, 0x3, 0x0)
        /home/xjrw/go/pkg/mod/github.com/filecoin-project/go-fil-markets@v0.2.7/storagemarket/impl/storedask/storedask.go:105 +0x38
github.com/filecoin-project/lotus/node/modules.NewStorageAsk(0x7fab5f972a50, 0xc00003e0d0, 0x2837b80, 0xc000538400, 0x281bc20, 0xc00047d460, 0xc00052245b, 0x3, 0x2823640, 0xc000ddc240, ...)
        /home/xjrw/scripts/lotus/node/modules/storageminer.go:302 +0x31e
reflect.Value.call(0x225bd80, 0x26198a0, 0x13, 0x247e5f9, 0x4, 0xc000dde380, 0x5, 0x5, 0x203000, 0xc0030ea700, ...)
        /usr/lib/golang/src/reflect/value.go:460 +0x5f6
reflect.Value.Call(0x225bd80, 0x26198a0, 0x13, 0xc000dde380, 0x5, 0x5, 0x10, 0x10, 0xc0004d86a0)
        /usr/lib/golang/src/reflect/value.go:321 +0xb4
github.com/filecoin-project/lotus/node.as.func2(0xc000dde380, 0x5, 0x5, 0xc000dde380, 0xc0004d99d0, 0x22cf360)
        /home/xjrw/scripts/lotus/node/options.go:140 +0xf0
reflect.callReflect(0xc000278420, 0xc0030ea940, 0xc0030ea928)
        /usr/lib/golang/src/reflect/value.go:549 +0x322
reflect.makeFuncStub(0x7fab5f972a50, 0xc00003e0d0, 0x2837b80, 0xc000538400, 0x281bc20, 0xc00047d460, 0xc00052245b, 0x3, 0x2823640, 0xc000ddc240, ...)
        /usr/lib/golang/src/reflect/asm_amd64.s:20 +0x42
reflect.Value.call(0x225bd80, 0xc000278420, 0x13, 0x247e5f9, 0x4, 0xc0003fc280, 0x5, 0x5, 0xc0030eac60, 0x51a598, ...)
        /usr/lib/golang/src/reflect/value.go:460 +0x5f6
reflect.Value.Call(0x225bd80, 0xc000278420, 0x13, 0xc0003fc280, 0x5, 0x5, 0xc0004c3380, 0xc0003fc280, 0x5)
        /usr/lib/golang/src/reflect/value.go:321 +0xb4
go.uber.org/dig.(*node).Call(0xc0003d23c0, 0x281a0c0, 0xc0004c3380, 0x282f0e0, 0x22d60a0)
        /home/xjrw/go/pkg/mod/go.uber.org/dig@v1.8.0/dig.go:665 +0x24a
go.uber.org/dig.paramSingle.Build(0x0, 0x0, 0x0, 0x282f0e0, 0x22d60a0, 0x281a0c0, 0xc0004c3380, 0x2126260, 0xc000670a20, 0x96, ...)
        /home/xjrw/go/pkg/mod/go.uber.org/dig@v1.8.0/param.go:245 +0x353
go.uber.org/dig.paramList.BuildList(0x282f0e0, 0x22dbfa0, 0xc0004eb860, 0xa, 0xa, 0x281a0c0, 0xc0004c3380, 0xc0030eb108, 0x553bac, 0x10, ...)
        /home/xjrw/go/pkg/mod/go.uber.org/dig@v1.8.0/param.go:201 +0xb8
go.uber.org/dig.(*node).Call(0xc0004eb900, 0x281a0c0, 0xc0004c3380, 0x282f0e0, 0x230e5a0)
        /home/xjrw/go/pkg/mod/go.uber.org/dig@v1.8.0/dig.go:656 +0xff
go.uber.org/dig.paramSingle.Build(0x0, 0x0, 0x0, 0x282f0e0, 0x230e5a0, 0x281a0c0, 0xc0004c3380, 0x23558a0, 0xc0002cad70, 0x94, ...)
        /home/xjrw/go/pkg/mod/go.uber.org/dig@v1.8.0/param.go:245 +0x353
go.uber.org/dig.paramList.BuildList(0x282f0e0, 0x221be40, 0xc000bd65c0, 0x4, 0x4, 0x281a0c0, 0xc0004c3380, 0x0, 0x0, 0x57c792, ...)
        /home/xjrw/go/pkg/mod/go.uber.org/dig@v1.8.0/param.go:201 +0xb8
go.uber.org/dig.(*Container).Invoke(0xc0004c3380, 0x221be40, 0x2619800, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/xjrw/go/pkg/mod/go.uber.org/dig@v1.8.0/dig.go:389 +0x2ba
go.uber.org/fx.(*App).executeInvokes(0xc000267080, 0x21adb40, 0xc0004d84a0)
        /home/xjrw/go/pkg/mod/go.uber.org/fx@v1.9.0/app.go:547 +0x36b
go.uber.org/fx.New(0xc000589920, 0x3, 0x3, 0xc0005a9b80)
        /home/xjrw/go/pkg/mod/go.uber.org/fx@v1.9.0/app.go:345 +0x4b8
github.com/filecoin-project/lotus/node.New(0x28082c0, 0xc00003e0d0, 0xc000219ec0, 0x5, 0x5, 0x0, 0x1, 0x0)
        /home/xjrw/scripts/lotus/node/builder.go:478 +0x488
main.glob..func9(0xc00040c8a0, 0x0, 0x0)
        /home/xjrw/scripts/lotus/cmd/lotus-storage-miner/run.go:104 +0x605
gopkg.in/urfave/cli%2ev2.(*Command).Run(0x43ecd60, 0xc00040c840, 0x0, 0x0)
        /home/xjrw/go/pkg/mod/gopkg.in/urfave/cli.v2@v2.0.0-20180128182452-d3ae77c26ac8/command.go:167 +0x6cf
gopkg.in/urfave/cli%2ev2.(*App).Run(0xc0005ac300, 0xc00000e080, 0x2, 0x2, 0x0, 0x0)
        /home/xjrw/go/pkg/mod/gopkg.in/urfave/cli.v2@v2.0.0-20180128182452-d3ae77c26ac8/app.go:261 +0x68c
main.main()
        /home/xjrw/scripts/lotus/cmd/lotus-storage-miner/main.go:81 +0x696
```

--- a/node/modules/storageminer.go
+++ b/node/modules/storageminer.go
@@ -298,6 +298,9 @@ func NewStorageAsk(ctx helpers.MetricsCtx, fapi lapi.FullNode, ds dtypes.Metadat
 	}
 
 	storedAsk, err := storedask.NewStoredAsk(namespace.Wrap(ds, datastore.NewKey("/deals/provider")), datastore.NewKey("latest-ask"), spn, address.Address(minerAddress))
+	if err != nil {
+		return nil, err
+	}
 	// Hacky way to set max piece size to the sector size
 	a := storedAsk.GetAsk(address.Address(minerAddress)).Ask
 	err = storedAsk.AddAsk(a.Price, a.Expiry-a.Timestamp, storagemarket.MaxPieceSize(abi.PaddedPieceSize(mi.SectorSize)))

